authors/apps/articles/tests/integration/test_article_api_endpoints.py

```diff
@@ -0,0 +1,46 @@
+from rest_framework import status
+
+from authors.apps.articles.tests import base_class
+
+
+class TestArticleAPIEndpoints(base_class.BaseTest):
+
+    def test_user_can_undo_a_like(self):
+        user, article = self.create_article_and_authenticate_test_user()
+        self.assertEqual(article.is_liked_by(user), False)
+        response = self.client.post(self.like_article_url(article.slug))
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(article.is_liked_by(user), True)
+        response = self.client.delete(self.like_article_url(article.slug))
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(article.is_liked_by(user), False)
+
+    def test_user_can_undo_a_dislike(self):
+        user, article = self.create_article_and_authenticate_test_user()
+        self.assertEqual(article.is_liked_by(user), False)
+        response = self.client.post(self.dislike_article_url(article.slug))
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(article.is_disliked_by(user), True)
+        response = self.client.delete(self.dislike_article_url(article.slug))
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(article.is_disliked_by(user), False)
+
+    def test_like_then_dislike_undoes_the_like(self):
+        user, article = self.create_article_and_authenticate_test_user()
+        article.liked_by.add(user)
+        self.assertEqual(article.is_liked_by(user), True)
+        self.assertEqual(article.is_disliked_by(user), False)
+        response = self.client.post(self.dislike_article_url(article.slug))
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(article.is_liked_by(user), False)
+        self.assertEqual(article.is_disliked_by(user), True)
+
+    def test_dislike_then_like_undoes_the_dislike(self):
+        user, article = self.create_article_and_authenticate_test_user()
+        article.disliked_by.add(user)
+        self.assertEqual(article.is_disliked_by(user), True)
+        self.assertEqual(article.is_liked_by(user), False)
+        response = self.client.post(self.like_article_url(article.slug))
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(article.is_disliked_by(user), False)
+        self.assertEqual(article.is_liked_by(user), True)
```

Comment:
what about simple likes, dislikes?

Reply:
Have you explored all possible test cases?

Reply:
Can a user like their own article?

Reply:
Yes, according to the story on `PT` they can